### DISCUSSION TITLE
How to build without setup.py after making changes in some part of the code on my local environement

### DISCUSSION THREAD

**Discussion by utkarsh0902311047** [pyg-team/pytorch_geometric#7605] [Q&A]
Hello, 
I used to run other small GitHub repositories. There, I download them to my environment, and if I make any changes can easily see their effect after running 'python setup.py install' and running the code.
Do you know how I can do the same with PyG?

Thank you

**Comment by wsad1** [comment]
If you are developing new features in `pyg` then the developing in pyg guide [here](https://github.com/pyg-team/pytorch_geometric/blob/master/.github/CONTRIBUTING.md#developing-pyg) should help. Specifically step 6 `Installing pyg in editable mode` should help.

**Reply by utkarsh0902311047** [reply]
I have cuda10.1 installed. 
With that, I am using torch1.8.1
But when I reach step 6, it installs torch 2.0.1, which starts to give me errors of "entry point."
Please can you suggest to me what are the specific cuda and torch versions required for editable mode?
Thank you

**Reply by wsad1** [reply]
[here](https://github.com/pyg-team/pytorch_geometric/blob/master/.github/CONTRIBUTING.md#developing-pyg) in steps 1 and 2 you can make sure you install the right versions of torch. 
Also I don't think cuda 10.1 is supported by the last two versions of torch, see here https://pytorch.org/get-started/previous-versions/

**Reply by utkarsh0902311047** [reply]
So I should install torch in cpu mode? Would that be ok?

**Reply by wsad1** [reply]
Yeah that should work.
